### ISSUE TITLE
Expose XR tracking state

### ISSUE THREAD

**Issue by ryantrem** [BabylonJS/BabylonNative#355]
We need to handle tracking lost and plumb it through somehow (it's not clear to me how WebXR exposes this, but from https://github.com/immersive-web/webxr/issues/243#issuecomment-454282109, it sounds like maybe it is just implicit from other state, like a null pose from the xr frame). Apps need to be able to present some kind of UI to help the user understand the tracking state and what they need to do to get tracking.

**Comment by ryantrem** [comment]
Talked with @RaananW about this and first we need to expose the tracking state in the Babylon.js layer (probably on `WebXRCamera`). This is tracked by bug: https://github.com/BabylonJS/Babylon.js/issues/9076.

Once that is in place, we need to update the Babylon Native XR contract (e.g. `XR.h`) and the platform specific implementations (e.g. `XR.cpp`, `XR.mm`) to follow the same rules (e.g. before we start tracking `XRFrame::getViewerPose` returns `null`, when tracking is lost `XRFrame::getViewerPose` returns a pose with the last known position but where `XRPose::emulatedPosition` returns `true`).

**Comment by CoPrez** [comment]
This work has been completed for ARCore and ARKit. A new issue has been filed to track exposing the states for OpenXR.

This PR exposes the tracking state in Babylon Native:
#504

This PR consumes the new tracking states in Babylon React Native:
[BabylonJS/BabylonReactNative/pull/122](https://github.com/BabylonJS/BabylonReactNative/pull/122)

This is the new issue filed for exposing the state from OpenXR:
#505
